pkg/cose/cose.go

```diff
@@ -0,0 +1,183 @@
+package cose
+
+import (
+	"crypto/ecdsa"
+	"crypto/sha256"
+	"crypto/sha512"
+	"crypto/x509"
+	"math/big"
+
+	"github.com/blocky/nitrite/pkg/attestation"
+	"github.com/blocky/nitrite/pkg/nitrite_error"
+	"github.com/fxamacker/cbor/v2"
+)
+
+type CoseHeader struct {
+	Alg interface{} `cbor:"1,keyasint,omitempty" json:"alg,omitempty"`
+}
+
+type CosePayload struct {
+	_           struct{} `cbor:",toarray"`
+	Protected   []byte
+	Unprotected cbor.RawMessage
+	Payload     []byte
+	Signature   []byte
+}
+
+type CoseSignature struct {
+	_           struct{} `cbor:",toarray"`
+	Context     string
+	Protected   []byte
+	ExternalAAD []byte
+	Payload     []byte
+}
+
+func (h *CoseHeader) AlgorithmString() (string, bool) {
+	switch h.Alg.(type) {
+	case string:
+		return h.Alg.(string), true
+	}
+	return "", false
+}
+
+func (h *CoseHeader) AlgorithmInt() (int64, bool) {
+	switch h.Alg.(type) {
+	case int64:
+		return h.Alg.(int64), true
+	}
+	return 0, false
+}
+
+func ExtractCosePayload(
+	data []byte,
+) (CosePayload, error) {
+	cose := CosePayload{}
+	err := cbor.Unmarshal(data, &cose)
+	if nil != err {
+		return CosePayload{}, nitrite_error.ErrBadCOSESign1Structure
+	}
+	return cose, nil
+}
+
+func VerifyCosePayload(
+	cose CosePayload,
+) error {
+	if cose.Protected == nil || len(cose.Protected) == 0 {
+		return nitrite_error.ErrCOSESign1EmptyProtectedSection
```

Comment:
Here you don't have an error to wrap, but could still use `fmt.Errorf` to avoid using sentinel errors